test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_Class.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.java_lang;
+
+import java.lang.constant.ClassDesc;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.util.Optional;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+/**
+ * This test Java.lang.Class API added in Java 12 and later version.
+ *
+ * new methods:
+ * - describeConstable()
+ * - descriptorString()
+ * - arrayType() 
+ * - componentType(): wrapper for Class.getComponentType(), not tested
+ */
+public class Test_Class {
+    public static Logger logger = Logger.getLogger(Test_Class.class);
+ 
+	/* descriptor strings */
+	String primitiveResult = "I";
+	String objectResult = "Ljava/lang/Object;";
+	String arrayResult = "[D";
+	String arrayResult2 = "[[[D";
+
+	/* test classes */
+	Class<?> primitiveTest = int.class;
+	Class<?> objectTest = new Object().getClass();
+	Class<?> arrayTest = new double[0].getClass();
+	Class<?> arrayTest2 = new double[0][][].getClass();
+
+	/*
+	 * Test Java 12 API Class.descriptorString()
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testClassDescriptorString() {
+		/* primitive type test */
+		logger.debug("testClassDescriptorString: Primitive test result is: " + primitiveTest.descriptorString() + " expected: " + primitiveResult);
+		Assert.assertTrue(primitiveTest.descriptorString().equals(primitiveResult));
+
+		/* ObjectType test */
+		logger.debug("testClassDescriptorString: Object test result is: " + objectTest.descriptorString() + " expected: " + objectResult);
+		Assert.assertTrue(objectTest.descriptorString().equals(objectResult));
+
+		/* ArrayType test */
+		logger.debug("testClassDescriptorString: Array test result is: " + arrayTest.descriptorString() + " expected: " + arrayResult);
+		Assert.assertTrue(arrayTest.descriptorString().equals(arrayResult));
+
+		/* ArrayType test 2 */
+		logger.debug("testClassDescriptorString: Array test 2 result is: " + arrayTest2.descriptorString() + " expected: " + arrayResult2);
+		Assert.assertTrue(arrayTest2.descriptorString().equals(arrayResult2));
+	}
+
+	/*
+	 * Test Java 12 API Class.describeConstable()
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testClassDescribeConstable() {
+		describeConstableTestGeneral("testClassDescribeConstable (primitive)", primitiveTest);
+		describeConstableTestGeneral("testClassDescribeConstable (object)", objectTest);
+		describeConstableTestGeneral("testClassDescribeConstable (array)", arrayTest);
+		describeConstableTestGeneral("testClassDescribeConstable (array 2)", arrayTest2);
+	}
+
+	private void describeConstableTestGeneral(String testName, Class<?> testClass) {
+		Optional<ClassDesc> optionalDesc = testClass.describeConstable();
```

Comment:
use `.orElseThrow()` and let the test code throw if the Optional is empty.  The test harness knows how to fail tests for unexpected exceptions.